common/txmgr/tracker.go

```diff
@@ -0,0 +1,248 @@
+package txmgr
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+
+	feetypes "github.com/smartcontractkit/chainlink/v2/common/fee/types"
+	txmgrtypes "github.com/smartcontractkit/chainlink/v2/common/txmgr/types"
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+const (
+	// defaultTTL is the default time to live for abandoned transactions
+	defaultTTL = 6 * time.Hour
+)
+
+// AbandonedTx is a transaction who's 'FromAddress' was removed from Confirmer's enabled addresses list
+type AbandonedTx[
+	CHAIN_ID types.ID,
+	ADDR types.Hashable,
+	TX_HASH types.Hashable,
+	BLOCK_HASH types.Hashable,
+	SEQ types.Sequence,
+	FEE feetypes.Fee,
+] struct {
+	id          int64
+	fromAddress ADDR
+	// fatalTime represents the time at which this transaction is to be marked fatal
+	fatalTime time.Time
+}
+
+// isValid returns false when it's past fatal time for this AbandonedTx
+func (atx *AbandonedTx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) isValid() bool {
+	return time.Now().Before(atx.fatalTime)
+}
+
+// Tracker tracks and finalizes abandoned transactions
+type Tracker[
+	CHAIN_ID types.ID,
+	ADDR types.Hashable,
+	TX_HASH types.Hashable,
+	BLOCK_HASH types.Hashable,
+	R txmgrtypes.ChainReceipt[TX_HASH, BLOCK_HASH],
```

Comment:
nit: is R short for ChainReceipt? What about CHAIN_RECEIPT or RECEIPT?

Reply:
Using `R` for consistency here. We should change this to `RECIEPT` at some point. We can make a ticket for some code improvements including refactoring how we pass the arguments list.